src/ipc/ipc4/helper.c

```diff
@@ -658,6 +659,61 @@ static int construct_config(struct ipc4_copier_module_cfg *copier_cfg, uint32_t
 	return IPC4_SUCCESS;
 }
 
+#if CONFIG_COMP_CHAIN_DMA
+int ipc4_create_mng_for_chain(struct ipc4_chain_dma *cdma)
+{
+	const struct sof_uuid uuid = {0x6a0a274f, 0x27cc, 0x4afb, {0xa3, 0xe7, 0x34,
+			0x44, 0x72, 0x3f, 0x43, 0x2e}};
+	const struct comp_driver *drv;
+	struct comp_dev *dev;
+
+	drv = ipc4_get_drv((uint8_t *)&uuid);
+	if (!drv)
+		return -EINVAL;
+
+	dev = drv->ops.create(drv, (const struct comp_ipc_config *)NULL,
+			      (const void *)cdma);
+	if (!dev)
+		return -EINVAL;
+
+	const uint32_t comp_id = IPC4_COMP_ID(cdma->primary.r.host_dma_id
+					      + IPC4_MAX_MODULE_COUNT, 0);
+	dev->ipc_config.id = comp_id;
+	dev->ipc_config.pipeline_id = cdma->primary.r.host_dma_id
+				      + IPC4_MAX_MODULE_COUNT;
+
+	return ipc4_add_comp_dev(dev);
+}
+
+int ipc4_chain_dma_state(struct comp_dev *dev, struct ipc4_chain_dma *cdma)
+{
+	const bool allocate = cdma->primary.r.allocate;
+	const bool enable = cdma->primary.r.enable;
+	int ret;
+
+	if (!dev)
+		return -EINVAL;
+	const struct comp_driver *drv = dev->drv;
+
+	if (allocate) {
+		if (enable)
+			ret = drv->ops.trigger(dev, COMP_TRIGGER_START);
```

Comment:
can we use the wrappers here rather than deref the ops pointers.

Reply:
Not sure if I understood you well, do you mean defining in chain_dma_state func something like this ? 
`
int (*trigger)(struct comp_dev *dev, int cmd);
trigger = &drv->ops.trigger;
`

Reply:
We should have static inline wrapper function for all DAI/DMA ops-> (as some are optional ops and we safely handle).
See component_ext.h i.e.
```c
/** See comp_ops::trigger */
static inline int comp_trigger(struct comp_dev *dev, int cmd)
{
	assert(dev->drv->ops.trigger);

	return (dev->is_shared && !cpu_is_me(dev->ipc_config.core)) ?
		comp_trigger_remote(dev, cmd) : dev->drv->ops.trigger(dev, cmd);
}
```